src/System.Net.NameResolution/src/System/Net/NameResolutionPal.Unix.cs

```diff
@@ -70,19 +70,15 @@ private static unsafe void ParseHostEntry(Interop.Sys.HostEntry hostEntry, bool
                     // is likely to involve extra allocations, hashing, etc., and so will probably be more expensive than
                     // this one in the typical (short list) case.
 
-                    var nativeAddresses = new Interop.Sys.IPAddress[hostEntry.IPAddressCount];
+                    Interop.Sys.IPAddress* nativeAddresses = stackalloc Interop.Sys.IPAddress[(int)hostEntry.AddressCount];
                     int nativeAddressCount = 0;
 
-                    Interop.Sys.addrinfo* addressListHandle = hostEntry.AddressListHandle;
-                    for (int i = 0; i < hostEntry.IPAddressCount; i++)
+                    Interop.Sys.IPAddress* addressHandle = hostEntry.AddressList;
+                    for (int i = 0; i < hostEntry.AddressCount; i++)
                     {
-                        Interop.Sys.IPAddress nativeIPAddress = default;
-                        int err = Interop.Sys.GetNextIPAddress(&hostEntry, &addressListHandle, &nativeIPAddress);
-                        Debug.Assert(err == 0);
-
-                        if (Array.IndexOf(nativeAddresses, nativeIPAddress, 0, nativeAddressCount) == -1)
+                        if (new Span<Interop.Sys.IPAddress>(addressHandle, nativeAddressCount).IndexOf(addressHandle[i]) == -1)
```

Comment:
I'm not understanding what this is doing.  It's creating a span around addressHandle, and then looking to see if that span contains addressHandle[i]?

Reply:
(The span should be created before the loop rather than each iteration.)